dev-support/k8s/deploy-notebook-controller.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+set -e
```

Comment:
How about `set -euo pipefail`

Reply:
good catch!
many shell scripts need to be set to `set -euo pipefail`.
Fix this problem in the next PR.

Reply:
@aylei  Thanks for your review!
@liuxunorg Ok, I will fix other shell scripts too.

Reply:
@lowc1012 Thanks!